File: typescript/getBatteryById/getBatteryById.ts
import { PublicAPIClient as APIClient } from "@novum-batteries/cloud-api-client";

const apiClient = new APIClient();

(async () => {
  try {
    await apiClient.login("E-MAIL", "PASSWORD");

    // In order to get the "id" of the require battery one can do that copying the id from the Service Center or querying your battery and get its id
    const battery = await apiClient.getBatteryById("DATABASE_ID");
    console.log(battery);
  } catch (e: any) {
    console.error(e.details);
  }
})();

File: python/getBatteryById/getBatteryById.py
from novum_api_client.client import NovumAPIClient

api = NovumAPIClient()

try:
    login = api.login("YOUR_EMAIL", "YOUR_PASSWORD")
    # In order to get the "id" of the require battery one can do that copying the id from the Service Center or querying your batteries and getting their id
    get_battery = api.get_battery_by_id("YOUR_BATTERY_ID")
    battery_name = get_battery.name
    print(f"Your battery named {battery_name} was fetched from our database.")

except:
    raise Exception("It was not possible to fetch your battery.")
